tests/e2e/tests/devfiles/JavaSpringBoot.spec.ts

```diff
@@ -33,30 +47,60 @@ const runTaskExpectedDialogue: string = 'Process 8080-tcp is now listening on po
 
 suite(`${stack} test`, async () => {
     suite(`Create ${stack} workspace`, async () => {
-        workspaceHandlingTests.createAndOpenWorkspace(stack);
+        // workspaceHandlingTests.createAndOpenWorkspace(stack);
+        test('Start Maven workspace using factory URL and vscode editor', async() => {
+            await dashboard.waitPage();
+            Logger.debug(`Fetching user kubernetes namespace, storing auth token by getting workspaces API URL.`);
+            await apiUrlResolver.getWorkspacesApiUrl();
+            await dashboard.clickCreateWorkspaceButton();
+            await createWorkspace.waitPage();
+            workspaceHandlingTests.setWindowHandle(await browserTabsUtil.getCurrentWindowHandle());
+            await createWorkspace.startWorkspaceUsingFactory(`https://github.com/che-samples/java-spring-petclinic/tree/devfilev2?che-editor=che-incubator/che-code/insiders&storageType=persistent`);
+            await browserTabsUtil.waitAndSwitchToAnotherWindow(workspaceHandlingTests.getWindowHandle(), TimeoutConstants.TS_IDE_LOAD_TIMEOUT);
+        });
         workspaceHandlingTests.obtainWorkspaceNameFromStartingPage();
         test('Register running workspace', async () => {
             CheReporter.registerRunningWorkspace(WorkspaceHandlingTests.getWorkspaceName());
         });
-        projectAndFileTests.waitWorkspaceReadiness(workspaceSampleName, workspaceRootFolderName, false);
+        test('Wait workspace readiness', async() => {
+            try {
+                await driverHelper.getDriver().wait(until.elementLocated(By.className('monaco-workbench')));
+            } catch (err) {
+                if ((err as Error).name === 'WebDriverError') {
```

Comment:
It's interesting to know which kind of 'WebDriverError' is handled by 3 seconds wait.

Can it be handled by adding timeout to command in try {} block? :
```
await driverHelper.getDriver().wait(until.elementLocated(By.className('monaco-workbench')), 3000);
```